### PR TITLE
net-fs/samba: Respect alternate SYSROOT in file copy workaround.

### PR DESCRIPTION
**Pull request by illiminable**

### CODE REVIEW

**Comment by Polynomial-C** [comment]
I'm not sure `${SYSROOT}` is correct here. Why not using `${EROOT}`?

File: net-fs/samba/samba-4.5.2.ebuild [hunk 1]
```diff
@@ -110,7 +110,7 @@ src_prepare() {
 	eapply "${WORKDIR}/patches/"
 
 	# ugly hackaround for bug #592502
-	cp /usr/include/tevent_internal.h "${S}"/lib/tevent/ || die
+	cp "${SYSROOT}"/usr/include/tevent_internal.h "${S}"/lib/tevent/ || die
```
Comment (SoapGentoo) [inline net-fs/samba/samba-4.5.2.ebuild]: please make this `${SYSROOT}${EPREFIX}` to be in line with line 182 of the ebuild

**Comment by SoapGentoo** [comment]
@illiminable please edit your COMMITMSGs to follow Gentoo format:
https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Commit_message_format
Also, squash commits into one, by learning how `git rebase` is your best friend
https://wiki.gentoo.org/wiki/Gentoo_Github#Rebasing_commits
You can squash + amend your COMMITMSGs all using `git rebase`

**Comment by Polynomial-C** [comment]
Can you please change the patches to apply against samba-4.5.4 and samba-4.6.0_rc1?

**Comment by mgorny** [comment]
@illiminable, ping. Are you still around?

**Comment by illiminable** [comment]
@mgorny Yes? Is there something still here for me to do? Do I need to rebase again?

**Comment by mgorny** [comment]
Yes, looks like it. Unless they were merged already, in which case you can just close the PR ;-).

CC-ing @chewi for cross expertise.

**Comment by chewi** [comment]
We haven't formally cemented what we're doing with SYSROOT yet but I think this tallies with our current line of thinking. The cross-compile case is somewhat academic until [bug #500090](https://bugs.gentoo.org/show_bug.cgi?id=500090) is fixed but this fix should work there too.

**Comment by mgorny** [comment]
@chewi, actually shouldn't SYSROOT include EPREFIX? After all, configure&co don't expect to append their --prefix to their --sysroot, do they?

**Comment by mgorny** [comment]
Anyway, this is somewhat a trivial code change but the exact details are unclear and it hasn't been merged for a very long time. I'm going to close it. If the maintainers finally agree on doing it, it can be easily done without a PR. Feel free to continue the discussion on a bug (and open one if it hasn't been created yet).

**Comment by illiminable** [comment]
So what is the status of bug #500090 ?

Aside just this change - I do at some point need to be able to do an ARM cross compile. I managed to do it previously through extreme brute force, but I really need something that is supported now.

eg. Hack the compiler args in flight https://chromium.googlesource.com/webports/+/master/ports/samba/cc_shim.sh

And - https://chromium.googlesource.com/webports/+/master/ports/samba/nacl.patch

Much of the python related issues though are now solved upstream.

**Pull request closed without merging**